packages/synapse-react-client/src/components/SetAccessRequirementCommonFields/SetAccessRequirementCommonFields.tsx

```diff
@@ -0,0 +1,309 @@
+} from '../../synapse-queries'
+import { SynapseClientError } from '../../utils/SynapseClientError'
+import EntitySubjectsSelector from '../EntitySubjectsSelector'
+import { SynapseSpinner } from '../LoadingScreen/LoadingScreen'
+import TeamSubjectsSelector from '../TeamSubjectsSelector'
+import { RadioGroup } from '../widgets/RadioGroup'
+
+export const EMPTY_SUBJECT_LIST_ERROR_MESSAGE =
+  'Please select at least one resource for this Access Requirement to be associated with.'
+export const UNSAVED_SUBJECTS_ERROR_MESSAGE = (
+  subjectsType: RestrictableObjectType,
+) => {
+  const idsText =
+    subjectsType === RestrictableObjectType.ENTITY ? 'Synapse' : 'Team'
+  const typeText =
+    subjectsType[0].toUpperCase() + subjectsType.slice(1).toLowerCase()
+  return `${idsText} IDs were specified but not added to the subjects list. Please either clear out the Add ${idsText} IDs textbox or click the Add ${pluralize(
+    typeText,
+  )} button.`
+}
+const NAME_HELP_TEXT =
+  "Enter access requirement name. This will also be used when sending notifications for expiring or revoked approval. For example, 'The approval for the name access requirement was revoked...'"
+
+const headerProps: Partial<TypographyProps> = {
+  variant: 'body1',
+  fontWeight: 700,
+}
+
+function getAccessType(subject: RestrictableObjectDescriptor) {
+  switch (subject.type) {
+    case RestrictableObjectType.ENTITY:
+      return ACCESS_TYPE.DOWNLOAD
+    case RestrictableObjectType.TEAM:
+      return ACCESS_TYPE.PARTICIPATE
+    default:
+      throw new Error(
+        `RestrictableObjectType ${subject.type} does not have an access type specified.`,
+      )
+  }
+}
+
+export type SetAccessRequirementCommonFieldsHandle = {
+  /* Allow the parent component to trigger saving the AR, so this may be embedded in an arbitrary modal. */
+  save: () => void
+}
+
+export type SetAccessRequirementCommonFieldsProps = {
+  /* Provided when creating a new AR */
+  subject?: RestrictableObjectDescriptor
+  /* Provided when editing an existing AR */
+  accessRequirementId?: string
+  /* Called when AR has been saved successfully */
+  onSave: (
+    accessRequirementId: string,
+    accessRequirementConreteType: ACCESS_REQUIREMENT_CONCRETE_TYPE,
+  ) => void
+  /* Called when error saving AR */
+  onError: () => void
+}
+
+export const SetAccessRequirementCommonFields = React.forwardRef(
+  function SetAccessRequirementCommonFields(
+    props: SetAccessRequirementCommonFieldsProps,
+    ref: React.ForwardedRef<SetAccessRequirementCommonFieldsHandle>,
+  ) {
+    const { subject, accessRequirementId, onSave, onError } = props
+
+    const isEditing = !subject
+
+    const [subjects, setSubjects] = useState<RestrictableObjectDescriptor[]>(
+      subject ? [subject] : [],
+    )
+    const [hasPendingSubjects, setHasPendingSubjects] = useState<boolean>(false)
+    const [subjectsError, setSubjectsError] = useState<string | null>(null)
+    const [name, setName] = useState<string>('')
```

Comment:
An empty string is allowed for the AR name, so allow any value here. The backend will validate whether the name is already in use (as in the case of an empty string).

SWC

https://github.com/Sage-Bionetworks/synapse-web-monorepo/assets/26949006/67b6b50f-bd5a-4d60-a45e-a5d739e8fccb

SRC

https://github.com/Sage-Bionetworks/synapse-web-monorepo/assets/26949006/9aea294e-f3b8-4cba-87e7-91fa724ce00e